lib/msal-node/test/client/ConfidentialClientApplication.spec.ts

```diff
@@ -229,206 +194,247 @@ describe("ConfidentialClientApplication", () => {
+            );
+        });
 
-    test('acquireTokenByClientCredential throws missingTenantIdError if "common", ""organization", or "consumers" was provided as the tenant id', async () => {
-        // @ts-ignore
-        const testProvider: msalCommon.IAppTokenProvider = () => {
-            // @ts-ignore
-            return new Promise<msalCommon.AppTokenProviderResult>((resolve) =>
-                resolve({
-                    accessToken: "accessToken",
-                    expiresInSeconds: 3601,
-                    refreshInSeconds: 1801,
-                })
+        test("ensures that developer-provided certificate is attached to client assertion", async () => {
+            const getClientAssertionSpy: jest.SpyInstance = jest.spyOn(
+                ClientApplication.prototype,
+                <any>"getClientAssertion"
             );
-        };
 
-        const appConfig: Configuration = {
-            auth: {
-                clientId: TEST_CONSTANTS.CLIENT_ID,
-                authority: TEST_CONSTANTS.DEFAULT_AUTHORITY, // contains "common"
-                clientAssertion: "testAssertion",
-            },
-        };
+            const cryptoKeys: CryptoKeys = new CryptoKeys();
+            config.auth.clientCertificate = {
+                thumbprint: cryptoKeys.thumbprint,
+                privateKey: cryptoKeys.privateKey,
+            };
+            delete config.auth.clientSecret;
+            const client: ConfidentialClientApplication =
+                new ConfidentialClientApplication(config);
 
-        const request: ClientCredentialRequest = {
-            scopes: TEST_CONSTANTS.DEFAULT_GRAPH_SCOPE,
-            skipCache: false,
-        };
+            const request: ClientCredentialRequest = {
+                scopes: TEST_CONSTANTS.DEFAULT_GRAPH_SCOPE,
+                skipCache: false,
+            };
 
-        const authApp = new ConfidentialClientApplication(appConfig);
-        authApp.SetAppTokenProvider(testProvider);
+            const authResult = (await client.acquireTokenByClientCredential(
+                request
+            )) as AuthenticationResult;
+            expect(authResult.accessToken).toEqual(
+                CONFIDENTIAL_CLIENT_AUTHENTICATION_RESULT.body.access_token
+            );
 
-        await expect(
-            authApp.acquireTokenByClientCredential(request)
-        ).rejects.toMatchObject(
-            createClientAuthError(ClientAuthErrorCodes.missingTenantIdError)
-        );
+            const clientAssertion: ClientAssertion = await getClientAssertionSpy
+                .mock.results[0].value;
+            expect(clientAssertion.assertion.length).toBeGreaterThan(1);
+            expect(clientAssertion.assertionType).toBe(
+                Constants.JWT_BEARER_ASSERTION_TYPE
+            );
+        });
     });
 
-    test("acquireTokenByClientCredential handles AuthErrors as expected", async () => {
```

Comment:
Deleted the test `acquireTokenByClientCredential handles AuthErrors as expected`.

This test doesn't accomplish anything